engine/glfw/lib/android/android_window.c

```diff
@@ -442,17 +442,21 @@ void glfwAndroidPollEvents()
    int events;
    struct android_poll_source* source;
 
-   int timeout = 0;
+   int timeoutMillis = 0;
    if (_glfwWin.iconified) {
-       timeout = 1000 * 3000;
+        timeoutMillis = 1000 * 3;
```

Comment:
according to Android documentation the timeout value is in ms not nanoseconds: https://developer.android.com/ndk/reference/group/looper#alooper_pollall

Reply:
Wow, ok, that's interesting!

Reply:
Yes! This maybe the main reason of ANR we see, actually. We have 5 sec top before ANR because of input ([and 10 sec for ANR because of BroadcastReceiver](https://developer.android.com/training/articles/perf-anr#anr)) and here we had a long 3000sec timeout:

Reply:
Yeah, using microseconds is common, but I guess someone didn't read the docs properly.

Q: Given that we are investigating ANR's which have a threshold of 500ms, should we really wait 3000ms then?

Reply:
it's a good question
I see messages like this:

>Input dispatching timed out (Waiting to send non-key event because the touched window has not finished processing certain input events that were delivered to it over 500.0ms ago. Wait queue length: 54. Wait queue head age: 8508.7ms.)

But in docs:
>No response to an input event (such as key press or screen touch events) within 5 seconds.

I don't see any ANRs where `Wait queue head age` is less that 5000ms. But I'm not sure if ANR happens after 500ms because `App has not finished processing certain input events that were delivered to it over 500.0ms ago` or it's just a message and it waits up to 5000ms like in docs